waspc/cli/Wasp/Cli/Command/Db/Migrate.hs

```diff
@@ -74,30 +74,28 @@ copyDbMigrationsDir ::
   Path' Abs (Dir ProjectRootDir) ->
   IO (Maybe String)
 copyDbMigrationsDir copyDirection waspProjectDir genProjectRootDir = do
-  let dbMigrationsDirInDbRootDir = [reldir|migrations|]
-
   -- Migration folder in Wasp source (seen by Wasp dev and versioned).
-  let dbMigrationsDirInWaspProjectDirAbs = waspProjectDir </> dbMigrationsDirInDbRootDir
+  let dbMigrationsDirInWaspProjectDirAbsPath = waspProjectDir </> dbMigrationsDirInWaspProjectDir
 
   -- Migration folder in the generated code.
-  let dbMigrationsDirInGenProjectDirAbs =
+  let dbMigrationsDirInGenProjectDirAbsPath =
         genProjectRootDir </> dbRootDirInProjectRootDir
           </> dbMigrationsDirInDbRootDir
 
-  let src =
+  let srcPathAbsDir =
         if copyDirection == CopyMigDirUp
-          then dbMigrationsDirInGenProjectDirAbs
-          else dbMigrationsDirInWaspProjectDirAbs
+          then SP.Path.toPathAbsDir dbMigrationsDirInGenProjectDirAbsPath
+          else SP.Path.toPathAbsDir dbMigrationsDirInWaspProjectDirAbsPath
 
-  let target =
+  let targetPathAbsDir =
         if copyDirection == CopyMigDirUp
-          then dbMigrationsDirInWaspProjectDirAbs
-          else dbMigrationsDirInGenProjectDirAbs
+          then SP.Path.toPathAbsDir dbMigrationsDirInWaspProjectDirAbsPath
+          else SP.Path.toPathAbsDir dbMigrationsDirInGenProjectDirAbsPath
```

Comment:
Ok, I see here what you mean by types not matching!

Yes, that is correct, this would be a problem because you are returning two different types. The question is, why would we do that, and should we rephrase what we want to do?
The thing is that here we actually don't know what source, or target, will exactly be -> they can be one folder in one case, or another folder in another case. And therefore it makes sense that type system complains. So it is us who need to make up our minds. And I think you did right, you said you know what, I actually don't care which folder it is. Ok, you went a step further and said I don't care about anything I will just turn it into FilePath and that is it :D, but that works perfectly here since this code is very localized and we need to convert it into FilePath any way just a few lines below, so I think this is perfect here. But, if you instead need to do some more complex stuff with `srcPathAbsDir` and `targetPathAbsDir` and instead wanted to have type system there to help you ensure they are absolute paths to a directory and using the filesystem conventions of underlying OS, maybe beause you are passing these to another function which takes strong path and expects such paths, then you could do that via `castDir` function (https://hackage.haskell.org/package/strong-path-1.1.1.0/docs/StrongPath.html#g:25), which casts directory to whatever you want, basically allowing you to "rename" the directory. In this case you would probalby want to rename both of these dirs to `Dir'` which effectively says they are both "any dir" and assuming that some other function that you need to pass these to takes `Dir'`, you would be good to go.

But, as I said, what you did here is actually perfect taking into account the circumstances, I wrote other stuff above just so you get an idea for some other use cases where it might be useful.

Reply:
Appreciate the explanation! Thanks for confirming the approach, and the pointer on `castDir` which will likely become super helpful in future! 👍🏻